### DISCUSSION TITLE
Quantification Table Format

### DISCUSSION THREAD

**Discussion by GeorgWa** [MannLabs/alphabase#89] [Ideas]
<table border=0 cellpadding=0 cellspacing=0 width=1029 style='border-collapse:
 collapse;table-layout:fixed;width:771pt'>
 <col width=87 span=3 style='width:65pt'>
 <col width=153 style='mso-width-source:userset;mso-width-alt:4906;width:115pt'>
 <col width=164 style='mso-width-source:userset;mso-width-alt:5248;width:123pt'>
 <col width=111 style='mso-width-source:userset;mso-width-alt:3541;width:83pt'>
 <col width=112 style='mso-width-source:userset;mso-width-alt:3584;width:84pt'>
 <col width=116 style='mso-width-source:userset;mso-width-alt:3712;width:87pt'>
 <col width=112 style='mso-width-source:userset;mso-width-alt:3584;width:84pt'>
 <tr height=21 style='height:16.0pt'>
  <td colspan=3 height=21 class=xl74 width=261 style='height:16.0pt;width:195pt'>Metadata
  Columns</td>
  <td colspan=2 class=xl74 width=317 style='width:238pt'>Index
  Columns</td>
  <td colspan=4 class=xl74 width=451 style='width:338pt'>Observation Columns</td>
 </tr>
 <tr height=21 style='height:16.0pt'>
  <td height=21 class=xl66 style='height:16.0pt'>protein</td>
  <td class=xl66>charge</td>
  <td class=xl66>peptide</td>
  <td class=xl69>level</td>
  <td class=xl69>quant_id</td>
  <td class=xl70>run1_channel1</td>
  <td class=xl70>run1_channel2</td>
  <td class=xl70>run2_channel1</td>
  <td class=xl70>run2_channel2</td>
 </tr>
 <tr height=21 style='height:16.0pt'>
  <td height=21 class=xl65 style='height:16.0pt'>P001</td>
  <td class=xl65 align=right>2</td>
  <td class=xl65>PEPTIDE</td>
  <td class=xl67>fragment</td>
  <td class=xl67>P001_PEPTIDE_b2</td>
  <td class=xl68 align=right>0,2</td>
  <td class=xl68 align=right>0,2</td>
  <td class=xl68 align=right>0,2</td>
  <td class=xl68 align=right>0,2</td>
 </tr>
 <tr height=21 style='height:16.0pt'>
  <td height=21 class=xl65 style='height:16.0pt'>P001</td>
  <td class=xl65 align=right>2</td>
  <td class=xl65>PEPTIDE</td>
  <td class=xl67>peptide_ms1</td>
  <td class=xl67>P001_PEPTIDE_ms1</td>
  <td class=xl68 align=right>4,3</td>
  <td class=xl68 align=right>4,3</td>
  <td class=xl68 align=right>5,3</td>
  <td class=xl68 align=right>4,3</td>
 </tr>
 <tr height=21 style='height:16.0pt'>
  <td height=21 class=xl65 style='height:16.0pt'>P002</td>
  <td class=xl65 align=right>3</td>
  <td class=xl65>DEPI</td>
  <td class=xl67>fragment</td>
  <td class=xl67>P002_DEPI_y4</td>
  <td class=xl68 align=right>4,3</td>
  <td class=xl68 align=right>4,3</td>
  <td class=xl68 align=right>5,3</td>
  <td class=xl68 align=right>4,3</td>
 </tr>
 <tr height=21 style='height:16.0pt'>
  <td height=21 class=xl65 style='height:16.0pt'>P002</td>
  <td class=xl65 align=right>3</td>
  <td class=xl65>DEPI</td>
  <td class=xl67>peptide_ms2</td>
  <td class=xl67>P002_DEPI_ms2</td>
  <td class=xl68 align=right>7,8</td>
  <td class=xl68 align=right>7,8</td>
  <td class=xl68 align=right>0,8</td>
  <td class=xl68 align=right>7,8</td>
 </tr>
 <tr height=21 style='height:16.0pt'>
  <td height=21 class=xl65 style='height:16.0pt'>P002</td>
  <td class=xl65 align=right></td>
  <td class=xl65></td>
  <td class=xl67>protein</td>
  <td class=xl67>P002</td>
  <td class=xl68 align=right>7,8</td>
  <td class=xl68 align=right>7,8</td>
  <td class=xl68 align=right>0,8</td>
  <td class=xl68 align=right>7,8</td>
 </tr>
</table>

### Metadata Columns
This group can contain general columns per quantification. It should allow to associate quantifications with the original PSMs. By default proteins, charge, peptide sequence, modification are included. 

_Implementation:_
There should be only a single generic implementation. They should be included based on regex matching the PSM columns.

### Index Columns
The index of the quantification table consists of the quantification level (**level**) and the quantification identifier (**quant_id**).

The different quantification levels can be defined by the conversion process. All quantities on the same quantification level must be comparable across identifications and observations. Common levels could be but are not limited to:

- fragment
- precursor
- peptide
- protein_group

_in principle arbitrary complex levels are possible e.g._
- precursor_MS1_isotope_0

The quantification ID should give an unique identifier for every quantity across observations. All quantifications across a quantification level should share the same format. Different quantification levels can have different formats.

### Observation Columns
Observation columns should start with the run name and contain the channel, in case multiple channels are used.

**Comment by GeorgWa** [comment]
Hi Everyone, thank you for the discussion today. I hope this proposal combines all of the requirements we have for it.

Please share If you would like to change or expand it, we can the update the definition.

**Comment by JuliaS92** [comment]
Hi everyone,
thank you @GeorgWa for writing this up!
My understanding was slightly different:
1. I thought we also wanted to use this format for protein level quantification, i.e. Type = protein(-group) and QuantID = PROT123;PROT123-3. How we want to map these back to the PSMs would be one question.
2. I thought the QuantID should look the same for all entries of one type, regradless of the loaded index columns. E.g. fragment = sequence_ion_..., whathever information is required to uniquely identify it. Independent of this additional columns can be loaded into the index, depending on what is required by the calling downstream application. E.g. PSM score or library RT for peptides, quantified peptides for proteins, etc.

Depending on where the data comes from, different columns need to be loaded to construct the QuantID and different index columns are available.

What do you think?

**Comment by ammarcsj** [comment]
Thanks a lot @GeorgWa  for writing this up. In my understanding, the QuantID would be part of the index, because it is a unique identifier for each row. Or maybe both quantification level columns in general could also be part of the index.
Concerning @JuliaS92's comments: 
1. Yes good question if we should also add the protein id in there, I guess it would make sense. One thing speaking against it, that on the protein level you can have the case that the mapping is non-unique, i.e. that potentially multiple proteins map to the same ion.
2. Yes I also understand it this way

**Comment by GeorgWa** [comment]
Hi, 

I also agree with both points. I think it's mostly a shortcoming of how I wrote it.

Regarding the first point. The example is not exhaustive, so I guess it makes sense to add a list of possible example quantification levels. The format itself can support any arbitrary level as long as quantities of the same level are comparable between identifications and runs. What do you think of:

> The different quantification levels can be defined by the conversion process. All quantities on the same quantification level must be comparable across identifications and observations. Common levels could be but are not limited to:
> - fragment
> - precursor
> - peptide
> - protein_group
>
> _in principle arbitrary complex levels are possible e.g._
> - precursor_MS1_isotope_0

Second point. Yes, agreed. Maybe defining it differenty will help. 

> The quantification ID should give an unique identifier for every quantity across observations. All quantifications across a quantification level should share the same format. Different quantification levels can have different formats.

**Comment by GeorgWa** [comment]
I've updated the description and the example. I think the naming was a bit confusing, maybe metadata makes the 'variable' nature of the additional columns more clear.

**Comment by JuliaS92** [comment]
I'd like to bring some options into play, to potentiall address the protein/peptide mapping issue and maybe also space issues:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SCHESS~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SCHESS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



Type | QuantID | Type_ID | Protein ids | Sequence | Peptides | Peptide ids
-- | -- | -- | -- | -- | -- | --
protein_group | PR1;PR2 | 1 | PR1;PR2 |   | PEPTIDE;TIDEPEP | 1;2
protein_group | PR3 | 2 | PR3 |   | PIPETED | 3
peptide | PEPTIDE | 1 | PR1 | PEPTIDE |   |  
peptide | TIDEPEP | 2 | PR1;PR2 | TIDEPEP |   |  
peptide | PIPETED | 3 | PR3 | PIPETED |   |  



</body>

</html>

1. As alterantive to the Quant ID we could use a Type_ID which is space saving and together with the Type still unique. Any additional info needed for processing would be in extra index columns (e.g. Sequence)
2. A protein IDs colkumn could just contain multiple ids for ambiguous mapping.
3. If we want to map from proteins to peptides without searching through all peptides we could either use the sequenes or the Type_IDs. The latter is how MaxQuant does it for referencing between it's output tables and I find it quite useful.
